climada_petals/hazard/landslide.py

```diff
@@ -37,43 +37,88 @@
 LOGGER = logging.getLogger(__name__)
 HAZ_TYPE = 'LS'
 
-def sample_events_from_probs(prob_matrix, n_years, dist='binom'):
-    """sample an event set for a specified representative time span from
-    a matrix with annual occurrence probabilities
-    Draws events from chosen distribution.
 
+def sample_events(prob_matrix, n_years, dist='binom'):
+    """
+    Sample yearly events for a specified time span (n_years) from
+    a matrix representing annual occurrence probabilities per grid cell 
+    (prob_matrix). The matrix will usually be provided throught the
+    Landslide.from_prob() method and refer to grid-cells and their annual 
+    sliding probabilities.
+    Events are drawn from the specified distribution (binomial or 
+    poisson). 
+    Returns n_year events for the whole grid, each event being representative
+    of all slides happening in 1 year throughout the entire grid-area.
+    
     Parameters
     ----------
-    prob_matrix : scipy.sparse.csr matrix
-        matrix where each entry has an annual probability [0,1] of occurrence of
-        an event
+    prob_matrix : np.array()
+        matrix where each entry has a probability [0,1] of occurrence of
+        an event. Shape of array can be any and is determined by hazard area 
+        and resolution of probability-source-file when called through 
+        Landslide.from_prob().
     n_years : int
-        the timespan of the probabilistic simulation in years
-    dist : str
-        distribution to sample from. currently 'binom' (default) and 'poisson'
-
+        the timespan of the probabilistic simulation in years. Each year will
+        result in 1 event.
+    dist : str, 'binom' or 'poisson'
+        distribution to sample from. Default is 'binom'.
+    
     Returns
     -------
-    ev_matrix : scipy.sparse.csr matrix
-        csr matrix with number of success (events) from sampling process per
-        grid cell
-
+    scipy.sparse.csr_matrix : 
+        shape is (shape(prob_matrix),n_years). Each non-zero entry is 1 (re-
+        ferring to a 'hit' in that year and grid cell.) 
+    
     See also
     --------
-    set_ls_prob(), scipy.stats.binom.rvs(), scipy.stats.poisson.rvs()
-
+    sample_event_from_probs()
+    Landslide.from_prob()
     """
-    LOGGER.info('Sampling landslide events for a %i year period', n_years)
 
-    ev_matrix = prob_matrix.copy()
+    events = [
+        sample_event_from_probs(prob_matrix, n_years=1, dist=dist)
+        for i in range(n_years)
+        ]
+    return sparse.csr_matrix(events)
+    
+    
+def sample_event_from_probs(prob_matrix, n_years, dist):
+    """sample for a specified representative time span  (n_years) from
+    a matrix with occurrence probabilities. Draws from a specified distribution.
```

Comment:
I still do not understand why two methods with the same signature are needed to sample events... But I suppose there is a good reason which is fine with me.

Reply:
i changed this now, indeed from n_years to n_samples in the signature of the 'sample_event_from_probs' methods, to indicate the difference between sampling once n_samples and sampling n_samples times one event